File: matrix_traversal/test/MatrixTest.java
import org.junit.Test;

public class MatrixTest {

    @Test
    public void testTraverse() throws Exception {
        assert 1 == Matrix.traverse(
                new int[][]{
                        {1}},
                1, 1);
        assert 1 == Matrix.traverse(
                new int[][]{
                        {1, 1},
                        {0, 1}},
                2, 2);
        assert 2 == Matrix.traverse(
                new int[][]{
                        {1, 1, 1},
                        {1, 0, 1},
                        {1, 1, 1}},
                3, 3);
        assert 11 == Matrix.traverse(
                new int[][]{
                        {1, 1, 1, 1},
                        {1, 1, 0, 1},
                        {1, 1, 1, 1},
                        {1, 1, 1, 1}},
                4, 4);
        assert 23 == Matrix.traverse(
                new int[][]{
                        {1, 1, 1, 1, 1},
                        {1, 1, 1, 0, 1},
                        {1, 1, 1, 1, 1},
                        {1, 1, 1, 1, 1}},
                4, 5);
        assert 10 == Matrix.traverse(
                new int[][]{
                        {1, 1, 1, 1},
                        {1, 1, 1, 1},
                        {1, 1, 1, 1}},
                3, 4);
        assert 4 == Matrix.traverse(
                new int[][]{
                        {1, 1, 1, 1},
                        {1, 1, 0, 1},
                        {1, 1, 1, 1}},
                3, 4);
    }
}

File: python_challenges/tests/matrix_test.py
__author__ = 'tilmann.bruckhaus'

import unittest
import matrix


class MatrixTest(unittest.TestCase):
    def test_traverse(self):
        self.assertEqual(0, matrix.Matrix([[0]]).traverse())
        self.assertEqual(1, matrix.Matrix([[1]]).traverse())
        self.assertEqual(1, matrix.Matrix([[1, 1],
                                           [0, 1]]).traverse())
        self.assertEqual(2, matrix.Matrix([[1, 1, 1],
                                           [1, 0, 1],
                                           [1, 1, 1]]).traverse())
        self.assertEqual(11, matrix.Matrix([[1, 1, 1, 1],
                                            [1, 1, 0, 1],
                                            [1, 1, 1, 1],
                                            [1, 1, 1, 1]]).traverse())
        self.assertEqual(23, matrix.Matrix([[1, 1, 1, 1, 1],
                                            [1, 1, 1, 0, 1],
                                            [1, 1, 1, 1, 1],
                                            [1, 1, 1, 1, 1]]).traverse())
        self.assertEqual(10, matrix.Matrix([[1, 1, 1, 1],
                                            [1, 1, 1, 1],
                                            [1, 1, 1, 1]]).traverse())
        self.assertEqual(4, matrix.Matrix([[1, 1, 1, 1],
                                           [1, 1, 0, 1],
                                           [1, 1, 1, 1]]).traverse())